apps/plea/models.py

```diff
@@ -639,3 +651,332 @@ class DataValidation(models.Model):
+        ("auditevent_api", "AuditEvent API event"),  # issue with an external component
+    )
+    EVENT_SUBTYPE_CHOICES = (
+        ("not_set", "Not set"),
+        ("success", "Success"),
+        ("EXT1", "External failure 1"),
+        ("EXT2", "External failure 2"),
+        ("case_strict_validation_failed", "Form Validation Error: Strict"),
+        ("case_loose_validation_failed", "Form Validation Error: Loose"),
+        ("case_invalid_missing_name", "Invalid Case: Missing Name"),
+        ("case_invalid_missing_urn", "Invalid Case: Missing URN"),
+        ("case_invalid_invalid_urn", "Invalid Case: Invalid URN"),
+        ("case_invalid_missing_dateofhearing", "Invalid Case: Missing date of hearing"),
+        ("case_invalid_name_too_long", "Invalid Case: Name too long"),
+        ("case_invalid_offencecode", "Invalid Case: Invalid offence code"),
+        ("case_invalid_courtcode", "Invalid Case: Invalid court code"),
+        ("case_invalid_not_in_whitelist", "Invalid Case: Not in whitelist"),
+        ("case_invalid_duplicate_offence", "Invalid Case: Duplicate offence"),
+        ("case_invalid_duplicate_urn_used", "Invalid Case: Duplicate URN already used (sent)"),
+        ("case_invalid_no_offences", "Invalid Case: No offences"),
+        ("result_invalid_duplicate_urn_used", "Invalid Result: Duplicate URN already used (sent)"),
+    )
+    IGNORED_CASE_FIELDS = [
+        "id",  # Django id is irrelavent
+        "auditevent",  # Case is saved before the auditevent so ignore it
+        "datavalidation",  # DataValidation objects are defunct
+    ]
+    IGNORED_RESULT_FIELDS = ["extra_data", "id"]
+    IGNORED_FORM_FIELDS = ["id"]
+    IGNORED_VALIDATOR_FIELDS = []
+
+    id = models.AutoField(
+        primary_key=True,
+        verbose_name="Audit Event ID")
+
+    case = models.ForeignKey(
+        Case,
+        verbose_name="Related case",
+        help_text="If there was a successful case loaded then it is related here",
+        null=True, blank=True)
+
+    event_type = models.CharField(
+        choices=EVENT_TYPE_CHOICES,
+        verbose_name="event type",
+        help_text="Identified the area of the application that the event happened in",
+        max_length=255)
+
+    event_subtype = models.CharField(
+        choices=EVENT_SUBTYPE_CHOICES,
+        verbose_name="reason for failure",
+        help_text="The specific reason for the event",
+        max_length=255)
+
+    event_trace = models.CharField(
+        max_length=4000,
+        verbose_name="error detail",
+        help_text="This detail about the reason for this event may be useful for developers to debug import issue",
+        blank=True, null=True)
+
+    event_data = HStoreField(
+        verbose_name="Event data",
+        help_text="If there was a failure and data fields were found, they are stored here to debug",
+        null=True, blank=True)
+
+    extra_data_hash = models.CharField(
```

Comment:
I don't quite understand the purpose of the `extra_data_hash`. We're dealing with changes small enough that we can just diff the dicts themselves and as far as I can tell we don't need to do any kind of lookup against previous hashes.